### PR TITLE
Add section on website modes

### LINKED ISSUE #100: pkgdown development mode
pkgdown offers 4 [development modes](https://pkgdown.r-lib.org/reference/build_site.html#development-mode): 

- `auto`
- `release`
- `devel`
- `unreleased`

At the moment, Epiverse-TRACE packages use either `auto`, or `release` (the default).

At the last development meeting, we discussed how `auto` could confuse users by providing two versions of the same site, possibly with conflicting information, or with information that doesn't match their installed version.
This lead us to call for a poll here to try and harmonize the use of these development mode across the project.

Here are the potential options we identified:

- keep `auto` but add visual elements (different colours, banner, etc.) to really insist on the difference between the release site and the development site
- use `dev` and always only offer the development site, with a prompt for users to update to the latest version if they detect mismatch with their existing version. Vignettes and documentation for their current version is still available locally via `help()` and `browseVignettes()`.
- use `unreleased` (only one website) until the package is on CRAN and then switch to `auto` (release + dev website). The rationale here is that we expect most users to install from CRAN and it would be good to direct them to the frozen release site, with an option to manually switch to the dev website if they installed from GitHub (which suggests they are probably more comfortable with R package lifecycle). This would also implement the first suggestion from this list.

### PR DESCRIPTION
**Pull request by jamesmbaazam**
This PR fixes #100 by adding a subsection under the "documentation" section to layout the rational for using the "unreleased" development mode for `pkgdown`.

### CODE REVIEW

File: principles.qmd [hunk 1]
```diff
@@ -119,6 +119,15 @@ Consistent, high quality, and accessible documentation is key to adopting softwa
 - Aim to provide translations for the most common languages
 - We aim to provide [automated blog posts for major and minor releases](https://epiverse-trace.github.io/blog.html#category=new-release). Code owners may occasionally decide to release patch blogs if the patch is critical.
 
+### Website modes
+
+Epiverse-TRACE packages use [`pkgdown`](https://pkgdown.r-lib.org/) to generate websites. `pkgdown` provides [options](https://pkgdown.r-lib.org/reference/build_site.html#development-mode) (in the `_pkgdown.yml` file) to generate different sites for the development and released versions of your package.
+
+When the package is not yet on CRAN, we use the `unreleased` mode. After the package is on CRAN, we switch to `auto`, which provides two versions of the website: `release` and `devel`. This is because we expect most users to install from CRAN and so we direct them to the frozen release site, with an option to manually switch to the `devel` website if they installed from GitHub.
```
Comment (joshwlambert) [inline principles.qmd:126]: Please provide an example package that is using `unreleased` and hyperlink it and the same for a package using `auto`. I always find it helps to explicitly see these setting being used in actual packages.

Comment (joshwlambert) [inline principles.qmd:126]: I would rephrase "the frozen release site" to something like "the site that is updated with every package version/CRAN release".

File: principles.qmd [hunk 2]
```diff
@@ -119,6 +119,15 @@ Consistent, high quality, and accessible documentation is key to adopting softwa
...
+
+We also provide text in the README to direct users to the appropriate site. Typical text for the README might be: "The package has two websites: one for `[the stable release version on CRAN](URL)`, and
```
Comment (joshwlambert) [inline principles.qmd:128]: Has this been discussed and agreed? I can add this to the packages I maintain if so.

**Review by joshwlambert — CHANGES_REQUESTED** [review]
Thanks for contributing to resolve issue #100 @jamesmbaazam!

I've left a couple of comments within the text.

I would also prefer if this was in it's own blueprints page. I think the information added is too specific to be considered a general principle. Instead I would like if there was a dedicated page for Epiverse-TRACE package websites. For now it can be a small page, just including this information, but gives us a place to expand on things like the website theme, etc.

File: principles.qmd (hunk 2 above)
Comment (jamesmbaazam) [reply principles.qmd:128]: > Has this been discussed and agreed? I can add this to the packages I maintain if so.

It was discussed in the issue comments and I am putting it here prospectively. We can agree to keep or remove it.

File: principles.qmd (hunk 1 above)
Comment (jamesmbaazam) [reply principles.qmd]: Thanks, Josh. I've incorporated your changes. Since the contents have moved, I'm unable to link the commits so, it'll be better to re-review.

Comment (jamesmbaazam) [reply principles.qmd]: Please see my previous on re-reviewing.

File: website.qmd [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+When the package is not yet on CRAN, we use the `unreleased` development mode. In this case, we use a visual element to highlight the package development version in red. An example is the [website](https://epiverse-trace.github.io/superspreading/) for `{superspreading}` and its associated `_pkgdown` [configuration](https://github.com/epiverse-trace/superspreading/blob/main/_pkgdown.yml#L12-L13).
+
+After the package is accepted on CRAN, we switch to `auto` mode, which provides two versions of the website: `release` and `devel`. Here, an example is the `{cfr}`package [website](https://epiverse-trace.github.io/cfr/), which is configured [here](https://github.com/epiverse-trace/cfr/blob/main/_pkgdown.yml#L5-L6). We use this configuration for the CRAN packages because we expect most users to install the package from CRAN and so we direct them to the CRAN site associate with the installed release. The additional `devel` option here allows users to manually switch to it if they installed the development version from GitHub.
```
Comment (joshwlambert) [inline website.qmd:7]:
```suggestion
After the package is accepted on CRAN, we switch to `auto` mode, which provides two versions of the website: `release` and `devel`. Here, an example is the `{cfr}`package [website](https://epiverse-trace.github.io/cfr/), which is configured [here](https://github.com/epiverse-trace/cfr/blob/main/_pkgdown.yml#L5-L6). We use this configuration for the CRAN packages because we expect most users to install the package from CRAN and so we direct them to the CRAN site associated with the current release. The additional `devel` option here allows users to manually switch to it if they installed the development version from GitHub.
```

Comment (joshwlambert) [inline website.qmd:7]: > The additional `devel` option here allows users to manually switch to it if they installed the development version from GitHub.

This sentence is unclear to me. I feel it would be more informative to fully explain that the default site under `auto` is `release` but there is a `devel` site also rendered and give the users instructions on how to access the `devel` site. From what I can see on the Epiverse packages that use `auto` there is no direct link to `devel` sites so users need to type `/dev` into the URL, but there may be a better way to do this that I'm unaware of.

File: website.qmd [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+We also provide text in package READMEs to direct users to the appropriate site. Typical text for the README might be: "The package has two websites: one for `[the stable release version on CRAN](URL)`, and another for `[the version in development](URL)`", where `URL` is the link to the respective site. See an example in the [`{EpiNow2}`] package [README](https://github.com/epiforecasts/EpiNow2/blob/main/README.Rmd#L125-L127).
```
Comment (joshwlambert) [inline website.qmd:9]: I would reword this from "We also provide" to something like "It is recommended to provide", as {EpiNow2} is not an Epiverse hosted package and I'm unaware of any Epiverse packages that state this in their READMEs (please point out packages that do this as it would be good to know).

**Review by joshwlambert — CHANGES_REQUESTED** [review]
Thanks for incorporating my previous requests @jamesmbaazam. I've requested a couple of minor changes in the text but overall it's looking good.

File: website.qmd [hunk 5]
```diff
@@ -0,0 +1,9 @@
+### Package website modes
+
+Epiverse-TRACE packages use the [`{pkgdown}` package](https://pkgdown.r-lib.org/) to generate websites. `{pkgdown}` provides [development modes](https://pkgdown.r-lib.org/reference/build_site.html#development-mode) (in the [`_pkgdown.yml` file](https://pkgdown.r-lib.org/articles/customise.html)) to generate different sites for the development and released versions of your package.
+
+When the package is not yet on CRAN, we use the `unreleased` development mode. In this case, we use a visual element to highlight the package development version in red. An example is the [website](https://epiverse-trace.github.io/superspreading/) for `{superspreading}` and its associated `_pkgdown` [configuration](https://github.com/epiverse-trace/superspreading/blob/main/_pkgdown.yml#L12-L13).
```
Comment (joshwlambert) [inline website.qmd:5]: Could expand the hyperlink to include "website for `{superspreading}`, but it's personal preference.

File: website.qmd [hunk 6]
```diff
@@ -0,0 +1,9 @@
+### Package website modes
+
+Epiverse-TRACE packages use the [`{pkgdown}` package](https://pkgdown.r-lib.org/) to generate websites. `{pkgdown}` provides [development modes](https://pkgdown.r-lib.org/reference/build_site.html#development-mode) (specified by a package's [`_pkgdown.yml` file](https://pkgdown.r-lib.org/articles/customise.html)) to generate different sites for the development and released versions of your package.
+
+When the package is not yet on CRAN, we use the `unreleased` development mode. In this case, we use a visual element to highlight the package development version in red. An example is the [website](https://epiverse-trace.github.io/superspreading/) for `{superspreading}` and its associated `_pkgdown` [configuration](https://github.com/epiverse-trace/superspreading/blob/main/_pkgdown.yml#L12-L13).
```
Comment (jamesmbaazam) [inline website.qmd:5]:
```suggestion
When the package is not yet on CRAN, we use the `unreleased` development mode. In this case, we use a visual element to highlight the package development version in red. An example is the [website for `{superspreading}`](https://epiverse-trace.github.io/superspreading/) and its associated `_pkgdown` [configuration](https://github.com/epiverse-trace/superspreading/blob/main/_pkgdown.yml#L12-L13).
```

File: website.qmd (hunk 4 above)
Comment (jamesmbaazam) [reply website.qmd:9]: There's none but we need to start doing that so that users are not confused when they visit the website and it doesn't correspond to the CRAN version. I'll remove it and we can add it after a wider discussion.

File: website.qmd [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+
+After the package is accepted on CRAN, we switch to `auto` mode, which provides two versions of the website: `release` and `devel`. Here, an example is the `{cfr}`package [website](https://epiverse-trace.github.io/cfr/), which is configured [here](https://github.com/epiverse-trace/cfr/blob/main/_pkgdown.yml#L5-L6). We use this configuration for the CRAN packages because we expect most users to install the package from CRAN and so we direct them to the CRAN site associated with the current release. The additional `devel` option here allows users to manually switch to it if they installed the development version from GitHub.
+
+We also provide text in package READMEs to direct users to the appropriate site. Typical text for the README might be: "The package has two websites: one for `[the stable release version on CRAN](URL)`, and another for `[the version in development](URL)`", where `URL` is the link to the respective site. See an example in the [`{EpiNow2}`] package [README](https://github.com/epiforecasts/EpiNow2/blob/main/README.Rmd#L125-L127).
```
Comment (jamesmbaazam) [inline website.qmd:9]:
```suggestion
```

File: website.qmd (hunk 5 above)
Comment (jamesmbaazam) [reply website.qmd]: I've done that in [677d2aa](https://github.com/epiverse-trace/blueprints/pull/125/commits/677d2aadad9701379c257f8d9437d30c29456dae).

File: website.qmd [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+
+When the package is not yet on CRAN, we use the `unreleased` development mode. In this case, we use a visual element to highlight the package development version in red. An example is the [website for `{superspreading}`](https://epiverse-trace.github.io/superspreading/) and its associated `_pkgdown` [configuration](https://github.com/epiverse-trace/superspreading/blob/main/_pkgdown.yml#L12-L13).
+
+After the package is accepted on CRAN, we switch to `auto` mode, which provides two versions of the website: `release` and `devel`. Here, an example is the `{cfr}`package [website](https://epiverse-trace.github.io/cfr/), which is configured [here](https://github.com/epiverse-trace/cfr/blob/main/_pkgdown.yml#L5-L6). We use this configuration for the CRAN packages because we expect most users to install the package from CRAN and so we direct them to the CRAN site associated with the current release. The additional `devel` option here allows users to manually switch to it if they installed the development version from GitHub.
```
Comment (jamesmbaazam) [inline website.qmd:7]:
```suggestion
After the package is accepted on CRAN, we switch to `auto` mode, which provides two versions of the website: `release` and `devel`. Here, an example is the `{cfr}`package [website](https://epiverse-trace.github.io/cfr/), which is configured [here](https://github.com/epiverse-trace/cfr/blob/main/_pkgdown.yml#L5-L6). We use this configuration for the CRAN packages because we expect most users to install the package from CRAN and so we direct them to the CRAN site associated with the current release. The additional `devel` option here allows users to manually switch to it (by adding `dev/` to the end of the website URL) if they installed the development version from GitHub. This version of the website will contain updates that are yet to be pushed to the CRAN version.
```

File: website.qmd (hunk 3 above)
Comment (jamesmbaazam) [reply website.qmd]: Does 3d55eb92362c03f1308630dc51d784f196e76181 clarify it?

Comment (joshwlambert) [reply website.qmd]: Yes! 👍

**Comment by joshwlambert** [comment]
A `.Rproj` file has been added in this PR. Not sure if this should be removed before merging.

**Comment by jamesmbaazam** [comment]
> A `.Rproj` file has been added in this PR. Not sure if this should be removed before merging.

Best not to introduce it, so I've removed it in [d782cf3](https://github.com/epiverse-trace/blueprints/pull/125/commits/d782cf38b855ab9e87bc0c63f0610bf2c28875e6).
